src/components/working-hours/change-log.tsx

```diff
@@ -1,36 +1,195 @@
-import { Table, TableBody, TableCell, TableHead, TableRow, Typography } from "@mui/material";
+import { ExpandMore } from "@mui/icons-material";
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography,
+} from "@mui/material";
+import {
+  AbsenceType,
+  Employee,
+  PerDiemAllowanceType,
+  WorkEvent,
+  WorkEventType,
+  WorkShiftChange,
+  WorkShiftChangeReason,
+  WorkShiftChangeSet,
+  WorkShiftHours,
+  WorkType,
+} from "generated/client";
+import { DateTime } from "luxon";
 import { useTranslation } from "react-i18next";
+import LocalizationUtils from "src/utils/localization-utils";
 
-function ChangeLog() {
+type Props = {
+  changeSets: WorkShiftChangeSet[];
+  workShiftDate: Date | undefined;
+  employees?: Employee[];
+  workShiftHours?: Record<WorkType, WorkShiftHours>;
+  workEvents?: WorkEvent[];
+};
+
+function ChangeLog({ changeSets, workShiftDate, employees, workShiftHours, workEvents }: Props) {
   const { t } = useTranslation();
 
+  if (!changeSets) {
+    return null;
+  }
+
+  // Sort changeSets by createdAt in descending order
+  changeSets.sort((a, b) => {
+    if (!a.createdAt || !b.createdAt) {
+      return 0;
+    }
+    const dateA = DateTime.fromJSDate(a.createdAt);
+    const dateB = DateTime.fromJSDate(b.createdAt);
+    return dateB.toMillis() - dateA.toMillis();
+  });
+
+  const getEmployeeFullName = (creatorId: string | undefined) => {
+    if (!employees) {
+      return creatorId;
+    }
+    const employee = employees.find((employee) => employee.id === creatorId);
+    if (!employee) {
+      return creatorId;
+    }
+
+    return `${employee.firstName} ${employee.lastName}`;
+  };
+
+  const formatChange = (oldValue: string | undefined, newValue: string | undefined) => {
+    return `${oldValue ?? "-"} -> ${newValue ?? "-"}`;
```

Comment:
The arrow right symbol might be better in this case as it differentiates better from the `-` that currently depicts empty value. Another option would be to define separate cases for value changing from or to 'undefined'.

```suggestion
    return `${oldValue ?? "-"} → ${newValue ?? "-"}`;
```